StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1201ElementsMustAppearInTheCorrectOrder.cs

```diff
@@ -296,7 +296,11 @@ private static void HandleMemberList(SyntaxNodeAnalysisContext context, Immutabl
 
                 if (index > nextIndex)
                 {
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, NamedTypeHelpers.GetNameOrIdentifierLocation(members[i + 1]), MemberNames[nextElementSyntaxKind], MemberNames[elementSyntaxKind]));
+                    // [Issue #3160] Added hardening here to make sure that this won't crash when working with invalid code.
+                    var nextElementMemberName = MemberNames.ContainsKey(nextElementSyntaxKind) ? MemberNames[nextElementSyntaxKind] : "<unknown>";
+                    var elementMemberName = MemberNames.ContainsKey(elementSyntaxKind) ? MemberNames[elementSyntaxKind] : "<unknown>";
```

Comment:
💡 I'd add the following to **DictionaryExtensions.cs**:

```csharp
public static TValue GetValueOrDefault<TKey, TValue>(this Dictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue)
  where TKey : notnull
{
  if (!dictionary.TryGetValue(key, out var value))
  {
    value = defaultValue;
  }

  return value;
}
```

Then these lines can be simplified:

```suggestion
                    var nextElementMemberName = MemberNames.GetValueOrDefault(nextElementSyntaxKind, "<unknown>");
                    var elementMemberName = MemberNames.GetValueOrDefault(elementSyntaxKind, "<unknown>");
```